### ISSUE TITLE
PackageName containing '@' character isn't display correctly in bitbucket-server

### ISSUE THREAD

**Issue by ThibautMarechal** [renovatebot/renovate#8803]
**What Renovate type, platform and version are you using?**

version: 24.42.1
platform: bitbucker-server

**Describe the bug**

I have a weird behavior using the platform bitbucket-server. Adding the zero-width character after the '@' fixed the issue.
![image](https://user-images.githubusercontent.com/20936586/108640832-20dce600-749c-11eb-9060-8cd8058855d9.png)
it display '@testing/react-autosuggest' but the real packageName is  '@type/react-autosuggest'. 
(@ testing is coming from the package above)
It only hapend when the packageName is in a link.
Could be fixed using the sanitizeMarkdown function in the util folder.

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [x] This is a really small bug, it does not need a reproduction (think small typo)
- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**
Draft PR: https://github.com/renovatebot/renovate/pull/8795

**Comment by rarkins** [comment]
What is the markdown code?

Also please expand your screenshot to include the full PR. e.g. you write "coming from the package above" but I don't see any package above.

**Comment by ThibautMarechal** [comment]
Bitbucket preview:
![image](https://user-images.githubusercontent.com/20936586/108682441-92468400-74f0-11eb-98e9-33529ce69076.png)
Markdown source:
![image](https://user-images.githubusercontent.com/20936586/108682855-20226f00-74f1-11eb-813b-831088d8f18a.png)

The sanitize function is adding `&#8203;` after the '@' and fix the issue

**Comment by rarkins** [comment]
So only the first `@` is escaped?

That's also a very strange bug if Bitbucket's HTML is displaying something completely different from the markdown.

Can you reproduce this is a public bitbucket.org repo?

**Comment by ThibautMarechal** [comment]
Sadly, it's a private repository 😕I don't have the time to setup a repository on bitbucket.org and configure renovate on it. 
It's only happening when the package has a link. It's not only the first one. 
Another example: 
![image](https://user-images.githubusercontent.com/20936586/108684542-519c3a00-74f3-11eb-93e2-53c2c163fa38.png)

**Comment by ThibautMarechal** [comment]
I did a simple snapshot test in the PR to validate the zero-width character is well added after the @ for packages with link. Let

**Comment by rarkins** [comment]
Can you reproduce it (privately) when there's only a single dependency being updated in a PR? Also don't worry about bitbucket.org - I misread which platform it is. Server and Cloud sometimes share code for frontend stuff like this but the majority of the codebase is separate, so it may not apply.

**Comment by ThibautMarechal** [comment]
When there is a single dependency, bitbucket-server display the correct text.

**Comment by rarkins** [comment]
So it requires two packages in a table, and both must have links? Or the problem occurs if the first has a link and the second does not?

Can you create the minimum PR possible that reproduces it, and copy and paste the entire markdown here using a code block so we can see it?

**Comment by ThibautMarechal** [comment]
To be clear, there is an issue in the markdown component from bitbucket UI (`@` is used to tag users, it's probably because of that). But we can prevent that bug from happening in renovate. 
Renovate already escaped de '@' here: https://github.com/renovatebot/renovate/blob/master/lib/workers/pr/body/updates-table.ts#L65, but only when starting with `@`. (regex `/^@/`). It's not matching that : `[@scope/package](link)`
I wanted just to remove the start condition but I found a function in the util folder that exactly does that. So to reduce duplicated code I use it

You can take my branch for that PR https://github.com/renovatebot/renovate/pull/8795/files. remove my fix and run the test that will fail: 
![image](https://user-images.githubusercontent.com/20936586/108703706-448b4500-750b-11eb-9d8b-f69e581f512b.png)

If you want the branch with the failing test (without the fix): https://github.com/ThibautMarechal/renovate/tree/withPackageNameBug

**Comment by aravindparappil46** [comment]
I'm guessing my qualm is related to this issue, hence commenting here!

I can confirm that in BitBucket Cloud (with self-hosted `renovate 24.89.2`), the code for zero-width character is being printed in the Onboarding PR for packages that have an @ in their name
![image](https://user-images.githubusercontent.com/43161107/111888977-c742db00-89b7-11eb-895d-a61ff169efc1.png)

I'm not facing the other issue mentioned by @ThibautMarechal where part of the package name is getting picked up from the package above.

**Comment by rarkins** [comment]
I think best to raise as a separate issue.

**Comment by JamieMagee** [comment]
@ThibautMarechal I can't comment on the issue, as I'm not a BitBucket user, but the fix looks reasonable (replacing some custom `string.replace` calls with a pre-existing helper function).

**Comment by Mause** [comment]
This is still an issue unfortunately, on Bitbucket Server especially

**Comment by ThibautMarechal** [comment]
Seems to be fixed with bitbucket server 7.17.5

**Comment by Mause** [comment]
@ThibautMarechal Renovate does aim to support all versions of Bitbucket Server that are within support though, and the one I'm using is quite far behind

**Comment by viceice** [comment]
@Mause We try to support as much versions as possible, but we don't add new features for out-of-support versions. We also only fix bugs for those versions when they are easy to fix and maintain.

**Comment by viceice** [comment]
Wrong, we only support not EOL version, so we only support bitbucket server 7.2+

- https://docs.renovatebot.com/modules/platform/bitbucket-server/#supported-versions
- https://confluence.atlassian.com/support/atlassian-support-end-of-life-policy-201851003.html#AtlassianSupportEndofLifePolicy-BitbucketServer

**Comment by viceice** [comment]
Reducing priority, as it's fixed in newer bitbucket server versions.
